subsys/net/lib/tls_credentials/tls_credentials.c

```diff
@@ -197,3 +197,26 @@ int tls_credential_delete(sec_tag_t tag, enum tls_credential_type type)
 
 	return ret;
 }
+
+#if defined(CONFIG_TLS_CREDENTIAL_KEYGEN)
+int tls_credential_keygen(sec_tag_t tag, enum tls_credential_keygen_type type,
+			  void *key, size_t *keylen)
+{
+	/* The volatile backend does not support keygen. */
+	return -ENOTSUP;
+}
+
+bool tls_credential_can_keygen(enum tls_credential_keygen_type type)
+{
+	/* The volatile backend does not support keygen. */
+	return false;
+}
+#endif /* defined(CONFIG_TLS_CREDENTIAL_KEYGEN)	*/
+
+#if defined(CONFIG_TLS_CREDENTIAL_CSR)
+int tls_credential_csr(sec_tag_t tag, char *dn, void *csr, size_t *csr_len)
+{
+	/* The volatile backend does not support CSR generation. */
+	return -ENOTSUP;
+}
+#endif /* defined(CONFIG_TLS_CREDENTIAL_CSR) */
```

Comment:
nit, GH complains about missing new line at the end of the line, here and in some other files.